### PR TITLE
GH-153: [feature] Create resend verification email endpoint

### PR DESCRIPTION
**Pull request by patrickmac3**
# GH-153 - Create Resend Verification Email Endpoint

## Description
This PR adds functionality for sending verification email to users to confirm their email. It includes an endpoint, services to use the keycloakapiclient for email sending. Unit tests for the controller are provided to ensure functionality. 

## Changes
- **Controller**: Added endpoints to assign badges (`PUT /auth/send-verification-email`) 
- **Services**: Added logic for using the Keycloak client to send the verification email
- **Client**: Added method for calling the keycloak endpoint for sending verification email. 
- **Tests**: Added unit tests for the controller/endpoint method.

### CODE REVIEW

File: Microservices/user-service/src/main/java/app/sportahub/userservice/controller/auth/AuthController.java [hunk 1]
```diff
@@ -44,5 +46,14 @@ public LoginResponse loginUser(@RequestBody LoginRequest loginRequest) {
...
     }
+
+    @PutMapping("/send-verification-email")
+    @ResponseStatus(HttpStatus.OK)
+    @Operation(summary = "send verification email",
+            description = "Sends a verification email to the specified user. This email contains a link or token that the user must use to confirm their email address and complete the registration."
+    )
+    public void sendVerificationEmail(@RequestBody Map<String, Object> payload) {
+        authService.sendVerificationEmail(payload.get("email").toString());
+    }
```
Comment (walidoow) [inline Microservices/user-service/src/main/java/app/sportahub/userservice/controller/auth/AuthController.java:57]: Can you use a response model under the /dto/response for the request body? Validate using Jakarta please.

File: Microservices/user-service/src/main/java/app/sportahub/userservice/service/auth/AuthServiceImpl.java [hunk 2]
```diff
@@ -88,11 +89,17 @@ public LoginResponse loginUser(LoginRequest loginRequest) {
...
         return Mono.from(keycloakApiClient.refreshToken(tokenRequest.refreshToken())).map(jsonNode -> {
             String refreshToken = jsonNode.get("refresh_token").asText();
             String accessToken = jsonNode.get("access_token").asText();
             return new TokenResponse(accessToken, refreshToken);
         }).block();
     }
+
+    @Override
+    public void sendVerificationEmail(String email) {
+        User user = userRepository.findUserByEmail(email).orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "User not found"));
```
Comment (walidoow) [inline Microservices/user-service/src/main/java/app/sportahub/userservice/service/auth/AuthServiceImpl.java:101]: In here, use the custom exception we have created for that purpose. Refer to this [link](https://github.com/SOEN490-SportsApp/SportsApp/blob/50d5dd3eb37053f5930247c58b1a57751bc5a4e5/Microservices/user-service/src/main/java/app/sportahub/userservice/service/user/UserServiceImpl.java#L66-L67) for an example use of the exception 👍

File: Microservices/user-service/src/main/java/app/sportahub/userservice/controller/auth/AuthController.java (hunk 1 above)
Comment (walidoow) [reply Microservices/user-service/src/main/java/app/sportahub/userservice/controller/auth/AuthController.java:57]: EDIT: @patrickmac3 I meant to use a request model un the `/dto/request` package

**Comment by walidoow** [comment]
Other than the comments I left above, everything works fine. Great work, I will accept this PR once you resolve the comments 👌

**Review by walidoow — APPROVED** [review]
Everything looks good to me! 🎉

**Review by Ziadsharkos — APPROVED** [review]
Yup, good work @patrickmac3

**Pull request merged**